commands/worldFlipper.js

```diff
@@ -12,48 +12,100 @@ const getGifEmbed = unit => new RichEmbed()
+    + '\n**Active Skill: **' + unit.EnSkillName + (unit.SkillCost ? ' **Cost: **' + unit.SkillCost : '')
+    + '\n' + unit.EnSkillDesc
+    + '\n**Rarity: **' + rarity)
     .addField('Ability 1', unit.EnAbility1, true)
     .addField('Ability 2', unit.EnAbility2, true)
     .addField('Ability 3', unit.EnAbility3, true)
     .setThumbnail(unit.SpriteURL)
-    .setFooter(unit.Role ? unit.Weapon + ' / ' + unit.Role : unit.Weapon);
+    .setFooter(footer);
 };
 
-const sendMessage = async (unit, message) => {
+const sendMessage = async (result, message) => {
+  const msg = await message.channel.send(getInfoEmbed(result.unit, result.versions));
+  let unit = result.unit;
   const artReaction = '🎨';
   const infoReaction = 'ℹ️';
   const gifReaction = '🎥';
+  // alt version reactions
+  const verReaction = [
+    '1️⃣',
+    '2️⃣',
+    '3️⃣',
+    '4️⃣',
+    '5️⃣',
+    '6️⃣',
+    '7️⃣',
+    '8️⃣',
+    '9️⃣',
+  ];
   const reactionExpiry = 30000;
   const filter = (reaction, user) => {
-    return [artReaction, infoReaction, gifReaction].includes(reaction.emoji.name) && user.id === message.author.id;
+    return [artReaction, infoReaction, gifReaction, ...verReaction].includes(reaction.emoji.name) && user.id === message.author.id;
   };
 
-  const msg = await message.channel.send(getInfoEmbed(unit));
-  await msg.react(artReaction);
-  await msg.react(infoReaction);
-  await msg.react(gifReaction);
+  // add reactions based on available data
+  if(unit.GifURL) {
+    await msg.react(gifReaction);
+  }
+  if(unit.SpriteURL) {
+    await msg.react(artReaction);
+  }
+  if(result.versions) {
+    for(let i = 0; i < result.versions.length; i++) {
+      await msg.react(verReaction[i]);
+    }
+  }
+
+  // create reaction collector
   const collector = msg.createReactionCollector(filter, { max: 10, time: reactionExpiry });
   collector.on('collect', r => {
     if (r.emoji.name === artReaction) {
+      // replace info with art
       msg.edit(getArtEmbed(unit));
     }
-    if (r.emoji.name === infoReaction) {
-      msg.edit(getInfoEmbed(unit));
-    }
     if (r.emoji.name === gifReaction) {
+      // replace info with gif
       msg.edit(getGifEmbed(unit));
     }
+    if(result.versions && verReaction.indexOf(r.emoji.name) > -1) {
+      // replace info with version based on reaction
+      // replace unit data
+      unit = result.versions[verReaction.indexOf(r.emoji.name)];
+      // edit message with new data and pass all version information
+      msg.edit(getInfoEmbed(unit, result.versions));
+      // clear current reaction and add them again
+      msg.clearReactions().then(async () => {
+        if(unit.GifURL) {
+          await msg.react(gifReaction);
+        }
+        if(unit.SpriteURL) {
+          await msg.react(artReaction);
+        }
+        if(result.versions) {
+          for(let i = 0; i < result.versions.length; i++) {
+            await msg.react(verReaction[i]);
+          }
```

Comment:
It was ages since I saw a real `for` loop, try using Array.forEach().

Reply:
Array.forEach() makes inconsistent reaction. for ... of doesn't return the keys, I need those.